From a6bac2d493a081dea6b1b04afe17028349b45aa4 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Fri, 14 Jul 2017 12:59:55 +0100
Subject: [PATCH 178/215] bcm2835-mmc: Fix DMA usage

The previous change ("bcm2835-mmc: Only claim one DMA channel")
used an incorrect variable, the effect of which was to prevent
DMA from being used at all. Fix that bug by using the right
variable.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 drivers/mmc/host/bcm2835-mmc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/bcm2835-mmc.c b/drivers/mmc/host/bcm2835-mmc.c
index 4fe8d1f..981db05 100644
--- a/drivers/mmc/host/bcm2835-mmc.c
+++ b/drivers/mmc/host/bcm2835-mmc.c
@@ -1354,14 +1354,14 @@ static int bcm2835_mmc_add_host(struct bcm2835_host *host)
 		if (ret == 0) {
 			host->dma_cfg_rx = cfg;
 
-			host->use_dma = true;
+			host->have_dma = true;
 		} else {
 			pr_err("%s: unable to configure DMA channel. "
-			       "Faling back to PIO\n",
+			       "Falling back to PIO\n",
 			       mmc_hostname(mmc));
 			dma_release_channel(host->dma_chan_rxtx);
 			host->dma_chan_rxtx = NULL;
-			host->use_dma = false;
+			host->have_dma = false;
 		}
 	}
 #endif
-- 
1.9.1

